[PyTorch] remove convenience RecordFunctionCallback interface

swolchok opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #48629 [PyTorch] Use plain old function pointer for RecordFunctionCallback
* **#48620 [PyTorch] remove convenience RecordFunctionCallback interface**

In preparation for storing bare function pointer (8 bytes)
instead of std::function (32 bytes).

Differential Revision: [D25132183](https://our.internmc.facebook.com/intern/diff/D25132183/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D25132183/)!

swolchok committed:
[PyTorch] remove convenience RecordFunctionCallback interface

In preparation for storing bare function pointer (8 bytes)
instead of std::function (32 bytes).

Differential Revision: [D25132183](https://our.internmc.facebook.com/intern/diff/D25132183/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D25132183/)!

[ghstack-poisoned]

swolchok committed:
apply clang-format, revert stray blank line on "[PyTorch] remove convenience RecordFunctionCallback interface"


In preparation for storing bare function pointer (8 bytes)
instead of std::function (32 bytes).

Differential Revision: [D25132183](https://our.internmc.facebook.com/intern/diff/D25132183/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D25132183/)!

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] remove convenience RecordFunctionCallback interface"


In preparation for storing bare function pointer (8 bytes)
instead of std::function (32 bytes).

Differential Revision: [D25132183](https://our.internmc.facebook.com/intern/diff/D25132183/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D25132183/)!

[ghstack-poisoned]

ilia-cher commented:
could you pls clarify the win here? it's not exactly a free win - this interface brings quite a lot of convenience to toss away without second thought

swolchok committed:
Update on "[PyTorch] remove convenience RecordFunctionCallback interface"


In preparation for storing bare function pointer (8 bytes)
instead of std::function (32 bytes).

Differential Revision: [D25132183](https://our.internmc.facebook.com/intern/diff/D25132183/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D25132183/)!

[ghstack-poisoned]

swolchok commented:
There are benchmark results on the FB diff at the top of the stack.

On Tue, Dec 8, 2020 at 3:29 PM ilia-cher <notifications@github.com> wrote:

> could you pls clarify the win here? it's not exactly a free win - this
> interface brings quite a lot of convenience to toss away without second
> thought
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/pytorch/pytorch/pull/48620#issuecomment-741226926>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAIK6CQVZ7QMMVDG3LS5J3ST2ZHNANCNFSM4UIF2R2A>
> .
>

swolchok committed:
Update on "[PyTorch] remove convenience RecordFunctionCallback interface"


In preparation for storing bare function pointer (8 bytes)
instead of std::function (32 bytes).

Differential Revision: [D25132183](https://our.internmc.facebook.com/intern/diff/D25132183/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D25132183/)!

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] remove convenience RecordFunctionCallback interface"


In preparation for storing bare function pointer (8 bytes)
instead of std::function (32 bytes).

Differential Revision: [D25132183](https://our.internmc.facebook.com/intern/diff/D25132183/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D25132183/)!

[ghstack-poisoned]
